client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,37 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * The values of this enum indicate if a ReaderGroup chooses Consumption based data retention for Streams it reads from.
```

Comment:
I hate to be pedantic, but this is public facing Javadoc, so I really want to nail it.
"this enum indicate if a ReaderGroup chooses Consumption based data retention"
This is not strictly accurate. The setting is actually on the stream so the reader group cannot "choose" consumption based retention. This only configures how this reader group interacts with the setting.

Let's make this explanation as precise as possible, and probably link to the Stream configuration setting so that users can locate it easily.

Reply:
fixed.